src/Domain/AddonServiceProvider.php

```diff
@@ -0,0 +1,88 @@
+<?php
+namespace GiveFBPT\Domain;
+
+use Give\Helpers\Hooks;
+use Give\ServiceProviders\ServiceProvider;
+
+use GiveFBPT\Domain\Helpers\SettingsPage;
+use GiveFBPT\Domain\SettingsPage as AddonSettingsPage;
+use GiveFBPT\Addon\Activation;
+use GiveFBPT\Addon\License;
+use GiveFBPT\Addon\Language;
+use GiveFBPT\Addon\ActivationBanner;
+
+/**
+ * Example of a service provider responsible for add-on initialization.
+ *
+ * @package     GiveFBPT\Addon
+ * @copyright   Copyright (c) 2020, GiveWP
+ */
+class AddonServiceProvider implements ServiceProvider {
+	/**
+	 * @inheritDoc
+	 */
+	public function register() {
+		give()->singleton( Activation::class );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function boot() {
+		// Load add-on translations.
+		Hooks::addAction( 'init', Language::class, 'load' );
+
+		is_admin()
+			? $this->loadBackend()
+			: $this->loadFrontend();
+	}
+
+
+	/**
+	 * Load add-on backend assets.
+	 *
+	 * @since 1.0.0
+	 * @return void
+	 */
+	private function loadBackend() {
```

Comment:
Make sure to clean this stuff up. It's a bunch of example code.

Reply:
For now, I'm choosing to leave this in place - since scope has not been ironed out entirely and it's likely that I will end up needing to load backend assets eventually. Once functionality is more settled, then I'll make sure to do a cleanup PR where I sweep out redundant or useless boilerplate.